packages/go_router/tool/run_tests.dart

```diff
@@ -9,27 +9,86 @@
 // ignore_for_file: avoid_print
 
 import 'dart:io';
+import 'package:io/io.dart' as io;
 import 'package:path/path.dart' as p;
+import 'package:yaml_edit/yaml_edit.dart';
 
 Future<void> main(List<String> args) async {
-  if (!Platform.isMacOS) {
-    print('This test can only be run on macOS.');
+  if (!Platform.isMacOS /* && !Platform.isWindows*/) {
+    print('This test can only be run on macOS' /*' or windows.'*/);
     exit(0);
   }
+  final p.Context ctx = p.context;
   final Directory packageRoot =
-      Directory(p.dirname(Platform.script.path)).parent;
+      Directory(ctx.dirname(_ensureTrimLeadingSeparator(Platform.script.path)))
+          .parent;
+
+  //copy from go_router/test_fixes to temp directory
+  final Directory testFixesTargetDir = await Directory.systemTemp.createTemp();
+
+  await _prepareTemplate(
+    ctx: ctx,
+    testFixesDir: ctx.join(packageRoot.path, 'test_fixes'),
+    testFixesTargetDir: testFixesTargetDir.path,
+    //Switching this to true, will depend on go_router v7.0.0, which will fail the golden test
+    testBadVersion: false,
+  );
+  final int pubGet = await _runProcess(
+    'dart',
+    <String>[
+      'pub',
+      'upgrade',
+    ],
+    workingDirectory: testFixesTargetDir.path,
+  );
+  if (pubGet != 0) {
+    exit(pubGet);
+  }
   final int status = await _runProcess(
     'dart',
     <String>[
       'fix',
       '--compare-to-golden',
     ],
-    workingDirectory: p.join(packageRoot.path, 'test_fixes'),
+    workingDirectory: testFixesTargetDir.path,
   );
 
   exit(status);
 }
 
+Future<void> _prepareTemplate({
+  required String testFixesDir,
+  required String testFixesTargetDir,
+  required p.Context ctx,
+  required bool testBadVersion,
+}) async {
+  await io.copyPath(testFixesDir, testFixesTargetDir);
+
+  final String pubspecYamlPath = ctx.join(testFixesTargetDir, 'pubspec.yaml');
+  final File targetPubspecPath = File(pubspecYamlPath);
+
+  final YamlEditor editor = YamlEditor(await targetPubspecPath.readAsString());
+  if (testBadVersion) {
+    editor.update(<String>['dependencies', 'go_router'], '7.0.0');
+  } else {
+    editor.update(
+      <String>['dependencies', 'go_router', 'path'],
```

Comment:
Is this else case needed?

Reply:
the if case is to demonstrate the failing behavior, the else case is the working one